kolibri/content/serializers.py

```diff
@@ -152,3 +182,47 @@ class Meta:
         )
 
         list_serializer_class = ContentNodeListSerializer
+
+class ContentNodeProgressListSerializer(serializers.ListSerializer):
+
+    def to_representation(self, data):
+
+        if not data:
+            return data
+
+        if 'request' not in self.context or not self.context['request'].user.is_facility_user:
+            progress_dict = {}
+        else:
+            user = self.context["request"].user
+            # Don't annotate topic progress as too expensive
+            progress_dict = get_topic_and_content_progress_fractions(data, user)
+
+        # Dealing with nested relationships, data can be a Manager,
+        # so, first get a queryset from the Manager if needed
+        iterable = data.all() if isinstance(data, Manager) else data
+
+        return [
+            self.child.to_representation(
+                item,
+                progress_fraction=progress_dict.get(item.content_id, 0.0),
+                annotate_progress_fraction=False
+            ) for item in iterable
+        ]
+
+class ContentNodeProgressSerializer(serializers.Serializer):
+
+    def to_representation(self, instance, progress_fraction=None, annotate_progress_fraction=True):
+        if progress_fraction is None and annotate_progress_fraction:
+            if 'request' not in self.context or not self.context['request'].user.is_facility_user:
+                # Don't try to annotate for a non facility user
+                progress_fraction = 0
+            else:
+                user = self.context["request"].user
```

Comment:
isn't this contradictory to the `if` above? 
how can we get user from `context['request']` if it's not available in the first place?